api/src/main/java/org/openmrs/module/appointments/service/impl/TeleconsultationAppointmentService.java

```diff
@@ -17,4 +37,51 @@ public String generateTeleconsultationLink(Appointment appointment) {
         }
         return new MessageFormat(tcServerUrl).format(new Object[] {appointment.getUuid()} );
     }
+
+    public AdhocTeleconsultationResponse generateAdhocTeleconsultationLink(String patientUuid, String provider) {
+        String identifierType = Context.getAdministrationService().getGlobalProperty(ADHOC_TC_ID);
+        Patient patient = patientService.getPatientByUuid(patientUuid);
+        PatientIdentifier identifier = patient.getIdentifiers().stream().filter(pi ->
+                        identifierType.equals(pi.getIdentifierType().getName()))
+                .findAny()
+                .orElse(null);
+        String id = generateRandomID();
```

Comment:
It feels more intuitive to write: String teleconID = null;
`teleconID = (identified != null)? identifier.getIdentifier(): generateRandomID()`

This way, it is is clear that: (a) In general we will use identifier.  And fallback is random().  (b) Random logic will execute only then, not everytime.

Reply:
Changed